### PR TITLE
[W5.4][W15-A2]Dong Shaocong

### PR DESCRIPTION
**Pull request by ShaocongDong**

### CODE REVIEW

File: src/seedu/addressbook/commands/HelpCommand.java [hunk 1]
```diff
@@ -18,9 +18,11 @@ public CommandResult execute() {
         return new CommandResult(
                 AddCommand.MESSAGE_USAGE
                 + "\n" + DeleteCommand.MESSAGE_USAGE
+                + "\n" + EditCommand.MESSAGE_USAGE
                 + "\n" + ClearCommand.MESSAGE_USAGE
                 + "\n" + FindCommand.MESSAGE_USAGE
                 + "\n" + ListCommand.MESSAGE_USAGE
+                + "\n" + SortCommand.MESSAGE_USAGE
                 + "\n" + ViewCommand.MESSAGE_USAGE
```
Comment (fanyiii) [inline src/seedu/addressbook/commands/HelpCommand.java]: Good job in updating Help command as well!

File: src/seedu/addressbook/data/person/Block.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    private String value;
+
+    private static final String BLOCK_NUMBER_CONSTRAINTS = "Block number is an integer between 0 to 10000";
+
+    /**
+     * construct the block object and validate the block number passed in
+     * @param inputNumber
+     * @throws IllegalValueException
+     */
+    public Block ( int inputNumber ) throws IllegalValueException {
```
Comment (fanyiii) [inline src/seedu/addressbook/data/person/Block.java]: Should be 'Block(int inputNumber) throws ...' instead

File: src/seedu/addressbook/data/person/Block.java [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+        if (!inputString.isEmpty()){
+            blockNumber = Integer.parseInt(inputString.trim());
+            valid = false;
+            if (!isValidBlockNumber(blockNumber)){
+                throw new IllegalValueException(BLOCK_NUMBER_CONSTRAINTS);
+            }
+            valid = true;
+            value = inputString.trim();
+        }
+        else {
```
Comment (fanyiii) [inline src/seedu/addressbook/data/person/Block.java]: else statement should not start on a new line

File: src/seedu/addressbook/data/person/Person.java [hunk 4]
```diff
@@ -14,26 +17,65 @@
     private Phone phone;
     private Email email;
     private Address address;
+    private Block block;
+    private Street street;
+    private Unit unit;
+    private Postal postal;
```
Comment (fanyiii) [inline src/seedu/addressbook/data/person/Person.java]: Since Person already has address field, which captures Block, Street etc, there is no need to declare them in the Person class again.

File: src/seedu/addressbook/data/person/ReadOnlyPerson.java [hunk 5]
```diff
@@ -13,6 +13,10 @@
     Phone getPhone();
     Email getEmail();
     Address getAddress();
+    Block getBlock();
+    Street getStreet();
+    Unit getUnit();
+    Postal getPostal();
```
Comment (fanyiii) [inline src/seedu/addressbook/data/person/ReadOnlyPerson.java]: Again, since Block, Street etc. are components of Address, there is no need to have them in ReadOnlyPerson.

File: src/seedu/addressbook/parser/Parser.java [hunk 6]
```diff
@@ -64,7 +67,7 @@ public Parser() {}
      * @param userInput full user input string
      * @return the command based on the user input
      */
-    public Command parseCommand(String userInput) {
+    public static Command parseCommand(String userInput) {
```
Comment (fanyiii) [inline src/seedu/addressbook/parser/Parser.java]: Why did you change it to a static method? A Parser object is actually created in Main class to execute this method (line 86).

File: src/seedu/addressbook/ui/Formatter.java [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+        return DISPLAYED_INDEX_OFFSET;
+    }
+
+    public static String getCommentLineFormatRegex () {
+        return COMMENT_LINE_FORMAT_REGEX;
+    }
+
+
+
+}
```
Comment (fanyiii) [inline src/seedu/addressbook/ui/Formatter.java]: I believe this is from LO5.10 SRP? Try to create a clean PR next time :)

File: test/java/seedu/addressbook/commands/EditCommandTest.java [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+    @Test
+    public void execute() throws IllegalValueException {
+
+    }
+
+    @Test
+    public void invalidEditValue() throws IllegalValueException {
+
+    }
+}
```
Comment (fanyiii) [inline test/java/seedu/addressbook/commands/EditCommandTest.java]: The JUnit tests are missing here but included in another PR instead. Next time try to make clean PRs and include all relevant code in your PR.

Comment (ShaocongDong) [reply test/java/seedu/addressbook/commands/EditCommandTest.java]: Thank you very much for your review! I will try not to mess up the PRs in the future! I will also consider your advice and try to make my PR mergeable next time :-)
